tests/unit/test_context.py

```diff
@@ -72,84 +88,129 @@ def test_explain():
     )
 
 
-def test_sql():
+@pytest.mark.parametrize(
+    "gpu",
+    [
+        False,
+        pytest.param(
+            True,
+            marks=(
+                pytest.mark.gpu,
+                pytest.mark.xfail(reason="create_table(gpu=True) doesn't work"),
+            ),
+        ),
+    ],
+)
+def test_sql(gpu):
     c = Context()
 
     data_frame = dd.from_pandas(pd.DataFrame({"a": [1, 2, 3]}), npartitions=1)
-    c.create_table("df", data_frame)
+    c.create_table("df", data_frame, gpu=gpu)
 
     result = c.sql("SELECT * FROM df")
-    assert isinstance(result, dd.DataFrame)
-    assert_frame_equal(result.compute(), data_frame.compute())
+    assert isinstance(result, dd.DataFrame if not gpu else dask_cudf.DataFrame)
+    dd.assert_eq(result, data_frame)
```

Comment:
Similar to `assert_correct_input` it might be valuable wrapping something that does this as a more common utility for pytests in a followup PR.